# Task 2: Visualizing fire scars through false color

**Github Repository:**

https://github.com/kpaya/EDS220-hmwk4

## Datasets description

The Thomas Fire, which burned over 280,000 acres in Ventura and Santa Barbara counties in December 2017, was one of California’s largest wildfires at the time. 

The following analysis uses a historical open-access dataset(1878-2023). The fire perimeter dataset contains spatial information for various fire incidents, including the fire perimeter of the Thomas Fire during 2017.

Source:

Data Gov. 2024. California fire perimeters (all). CAL FIRE. Retrieved Nov.14,2024 from https://catalog.data.gov/dataset/california-fire-perimeters-all-b3436

## 2. Fire perimeter data retrieval and selection

In [15]:
# Import libraries
import os
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import geopandas as gpd
import xarray as xr
import rioxarray as rioxr

In [23]:
# Read data
fp=os.path.join('/Users/kpaya/MEDS/EDS220/eds220-hmwk/EDS220-hmwk4/data/California_Fire_Perimeters_(all).shp')
fire_perimeter=gpd.read_file(fp)

## Data Exploration for fire_perimeter

In [ ]:
# Display the first few rows of the DataFrame to get an overview of the data

fire_perimeter.head(3)

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
0   2023    CA    CDF     SKU  WHITWORTH  00004808  2023-06-17  2023-06-17   
1   2023    CA    LRA     BTU     KAISER  00010225  2023-06-02  2023-06-02   
2   2023    CA    CDF     AEU    JACKSON  00017640  2023-07-01  2023-07-02   

   CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NA  \
0      5         1          1    5.72913     None       None   
1      5         1          1   13.60240     None       None   
2      2         1          1   27.81450     None       None   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}     None       None     2020   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}     None       None     2020   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}     None       None     2020   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...

In [ ]:
# Diplay columns of the DataFrame

fire_perimeter.columns

Index(['YEAR_', 'STATE', 'AGENCY', 'UNIT_ID', 'FIRE_NAME', 'INC_NUM',
       'ALARM_DATE', 'CONT_DATE', 'CAUSE', 'C_METHOD', 'OBJECTIVE',
       'GIS_ACRES', 'COMMENTS', 'COMPLEX_NA', 'IRWINID', 'FIRE_NUM',
       'COMPLEX_ID', 'DECADES', 'geometry'],
      dtype='object')

In [24]:
# Replace spaces with underscores and convert to lowercase

fire_perimeter.columns = fire_perimeter.columns.str.rstrip().str.replace(' ', '_').str.lower()

# Replace the column name 'year' explicitly to avoid '_'
fire_perimeter.columns = fire_perimeter.columns.str.replace(r'year_', 'year')

# Check results
fire_perimeter.columns

Index(['year', 'state', 'agency', 'unit_id', 'fire_name', 'inc_num',
       'alarm_date', 'cont_date', 'cause', 'c_method', 'objective',
       'gis_acres', 'comments', 'complex_na', 'irwinid', 'fire_num',
       'complex_id', 'decades', 'geometry'],
      dtype='object')

In [31]:
# Check type of each column

print(fire_perimeter.dtypes)

year             int64
state           object
agency          object
unit_id         object
fire_name       object
inc_num         object
alarm_date      object
cont_date       object
cause            int64
c_method         int64
objective        int64
gis_acres      float64
comments        object
complex_na      object
irwinid         object
fire_num        object
complex_id      object
decades          int64
geometry      geometry
dtype: object


In [32]:
# Check the data type of the dataframe 

print(type(fire_perimeter))

<class 'geopandas.geodataframe.GeoDataFrame'>


In [33]:
# Check the data type of the geometry column

print(type(fire_perimeter.geometry))

<class 'geopandas.geoseries.GeoSeries'>


In [21]:
# Print dataset's projection
print(f"{'The CRS of fire_perimeter is:':} {fire_perimeter.crs}")

The CRS of fire_perimeter is: epsg:3857


# Summary

The results of the preliminary exploration are the following:

Data Structure:

The dataset consists of 17 columns, with information such as fire year (year), state, agency, etc.

The geometry column contains spatial information (polygons representing the fire perimeters)

Column Names:

The column names were cleaned to replace spaces with underscores and convert them to lowercase for easier processing and consistency, (for example: YEAR_ was changed to year)

Data Types:

The dataset is a GeoDataFrame, as indicated by the type(fire_perimeter) output
The geometry column is a GeoSeries, which is a data structure for handling geometric shapes in geospatial analysis.

Coordinate Reference System (CRS):

The dataset uses the EPSG:3857 coordinate reference system (CRS), which is a projected coordinate system.

## Data Pre-processing


In [26]:
# Filter the fire perimeter for the 2017 Thomas Fire
thomas_fire = fire_perimeter[(fire_perimeter['fire_name'] == 'THOMAS') & (fire_perimeter['year'] == 2017)]

# Define the file path where the shapefile will be saved (adjust the path as needed)
output_path = 'data/thomas_fire_2017_boundary.shp'

# Export the filtered boundary to a shapefile
thomas_fire.to_file(output_path)


By exporting the thomas_fire_2017 boundary as a .shp file, it enables us to:

- use a standard vector-based spatial operations on the fire perimeter

- Overlay and analysis of both vector and raster